app/components/channel_info/index.js

```diff
@@ -0,0 +1,420 @@
+            this.enableRightButton(canUpdate);
+        }
+    };
+
+    onDisplayURLChangeText = (channelURL) => {
+        const {editing, onChannelURLChange} = this.props;
+        onChannelURLChange(channelURL);
+
+        if (editing) {
+            const {displayName, purpose, header} = this.props;
+            const canUpdate = this.canUpdate(displayName, channelURL, purpose, header);
+            this.enableRightButton(canUpdate);
+        }
+    };
+
+    onPurposeChangeText = (purpose) => {
+        const {editing, onPurposeChange} = this.props;
+        onPurposeChange(purpose);
+
+        if (editing) {
+            const {displayName, channelURL, header} = this.props;
+            const canUpdate = this.canUpdate(displayName, channelURL, purpose, header);
+            this.enableRightButton(canUpdate);
+        }
+    };
+
+    onHeaderChangeText = (header) => {
+        const {editing, onHeaderChange} = this.props;
+        onHeaderChange(header);
+
+        if (editing) {
+            const {displayName, channelURL, purpose} = this.props;
+            const canUpdate = this.canUpdate(displayName, channelURL, purpose, header);
+            this.enableRightButton(canUpdate);
+        }
+    };
+
+    scrollRef = (ref) => {
+        this.scroll = ref;
+    };
+
+    scrollToEnd = () => {
+        this.scroll.scrollToFocusedInput(findNodeHandle(this.lastText));
+    };
+
+    render() {
+        const {theme, editing, displayHeaderOnly, currentTeamUrl} = this.props;
+        const {displayName, channelURL, header, purpose} = this.props;
+        const {error, creating, updating} = this.props;
+        const {height, width} = Dimensions.get('window');
+        const fullUrl = currentTeamUrl + '/channels';
+        const shortUrl = getShortenedURL(fullUrl, 35);
+
+        const style = getStyleSheet(theme);
+
+        if (creating || updating) {
+            return (
+                <View style={{flex: 1}}>
+                    <StatusBar/>
+                    <Loading/>
+                </View>
+            );
+        }
+
+        let displayError;
+        if (error) {
+            displayError = (
+                <View style={[style.errorContainer, {width}]}>
+                    <View style={style.errorWrapper}>
+                        <ErrorText error={error}/>
+                    </View>
+                </View>
+            );
+        }
+
+        return (
+            <View style={{flex: 1}}>
+                <StatusBar/>
+                <KeyboardAwareScrollView
+                    ref={this.scrollRef}
+                    style={style.container}
+                >
+                    {displayError}
+                    <TouchableWithoutFeedback onPress={this.blur}>
+                        <View style={[style.scrollView, {height: height + (Platform.OS === 'android' ? 200 : 0)}]}>
+                            {!displayHeaderOnly && (
```

Comment:
can you explain when `diplayHeaderOnly` will be true and why?

Reply:
`displayHeaderOnly` is used for channels that are DMs or GMs.  From what I could tell on desktop you can only update the channel header. Here's what a DM with you looks like:

<img width="350" alt="screen shot 2017-11-24 at 9 30 06 am" src="https://user-images.githubusercontent.com/6182543/33214521-4b744fe6-d0fa-11e7-8494-42f206bf8a9e.png">

some possible name changes could be `editHeaderOnly` or `isDMorGM` to be a bit more clear

Reply:
Maybe change it to pass in the channel type? It's very likely that we're going to want to show different options here for other types of channels as well